src/policy/feerate.cpp

```diff
@@ -7,7 +7,7 @@
 
 #include <tinyformat.h>
 
-const std::string CURRENCY_UNIT = "LTC";
+const std::string CURRENCY_UNIT = "MONA";
 
 CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
```

Comment:
If value of `nFeePaid` is 10512000000000000(MAX_MONEY), it's overflow?
nFeePaid * 1000 = 
`10,512,000,000,000,000,000`
int64_t of max is 
`9,223,372,036,854,775,807`

So......how about adding the following?
`assert(nFeePaid<= MAX_MONEY);`

Personaly, I'd like to believe that maybe this isn't the case.

Reply:
Is this a good practice to use `asset()` here?
I suppose all Monacoin-core nodes will be aborted by assertion failed.
`assert()` in current code is checks for the runtime (compiler) limitations. Not for argument checks.

My alternative plan is to use `int128_t` and/or `uint128_t` in functions. WDYT?

Reply:
It's good.
I'm not very familiar with C++, but it looks like it says you have to use `BOOST`.
If you're putting in a new library, so......is it better to use `arith_uint256`?

Reply:
How about fix the type of `CAmount` ?

In `src/amount.h`:

```diff
  /** Amount in satoshis (Can be negative) */
- typedef int64_t CAmount;
+ typedef int128_t CAmount;
```

It looks this was discussed in [the bitcoin core team](http://gnusha.org/bitcoin-core-dev/2019-07-15.log).

> 11:16 < Raystonn> Normally I'd agree.  But upon testing a change of CAmount to  boost::multiprecision::int128_t, the floating point operations aren't even supported.  I could add them, but it could lead to dropped precision or accuracy some place where it matters simply by defining the operators.
> 11:16 < sipa> there is no need for 128 bit integers
> 11:16 < Raystonn> Right now no.

They didn't need it. And it may be worth to consider for us.

Reply:
Ooops, `s/int128_t/__int128_t/` . AFAIK, `__int128_t` is GCC extension, supported by Clang also.

Reply:
Another plan. I guess this plan is safer.

```diff
-         nSatoshisPerK = nFeePaid * 1000 / nSize;
+         nSatoshisPerK = (CAmount)((__int128_t)nFeePaid * 1000 / nSize);
```

Reply:
@cryptcoin-junkey 
I disagree with setting CAmount to `int128_t`.
The impact is toooooooooo great.

I agree to an another plan.